src/librustc/traits/project.rs

```diff
@@ -1240,7 +1237,7 @@ fn confirm_impl_candidate<'cx, 'gcx, 'tcx>(
         // just return TyError.
         debug!("confirm_impl_candidate: no associated type {:?} for {:?}",
                assoc_ty.item.name,
-               obligation.predicate.trait_ref);
+               obligation.predicate.trait_ref(tcx));
```

Comment:
For debug printing (here and elsewhere) I'd just let it print the value that used to have a `trait_ref` field.